File: twoSum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map <Integer, Integer> seen = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
          int comp = target - nums[i];
          
          if(seen.containsKey(comp)){
            return new int[] {seen.get(comp), i};
          } else {
            seen.put(nums[i], i);
          }
        }
        throw new IllegalArgumentException("No Match");
    }
    
    public static void main(String[] args){
      Solution solution = new Solution();
      
      int[] nums = {2,7,11,15};
      int target = 9;
      
      int[] result = solution.twoSum(nums, target);
      
      System.out.println(Arrays.toString(result));
    }
}

File: python/twosum.py
from typing import List

nums = [3,2,3]
target = 6

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        thisdict = {}
        for i in range(len(nums)):
            comp = target - nums[i]
            if comp in thisdict:
              return [thisdict[comp], i]
            else:
              thisdict[nums[i]] = i

solver = Solution()
solver.twoSum(nums,target)
